src/app/app.component.ts

```diff
@@ -1,10 +1,59 @@
 import { Component } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { PARAMETERS } from '@angular/core/src/util/decorators';
+import{HttpModule} from '@angular/http';
+import { DatePipe } from '@angular/common';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
-  title = 'app';
+
+  rForm: FormGroup;
+  post: any;
+  fname: string = '';
+  lname: string ='';
+  branch: string ='';
+  yop: number;
+  place: string ='';
+  mobilenum: number;
+  dob: number;
+  regnum: number;
+  emailid: string ='';
+  description: string = '';
+  titleAlert: string = 'Please enter a valid name ex: Raj size 3-32 char';
+
+  constructor(private fb: FormBuilder) {
+    this.rForm = fb.group({
+      'fname': [null, Validators.compose([Validators.required,Validators.pattern('^[a-zA-Z]+$'), Validators.minLength(3), Validators.maxLength(32)])],
+      'lname': [null, Validators.compose([Validators.required,Validators.pattern('^[a-zA-Z]+$'), Validators.minLength(3), Validators.maxLength(32)])],
+      'place': [null, Validators.compose([Validators.required,Validators.pattern('^[a-zA-Z]+$'), Validators.minLength(3), Validators.maxLength(32)])],
+      'branch': [null, Validators.compose([Validators.required,Validators.pattern('^[a-zA-Z]+$'), Validators.minLength(3), Validators.maxLength(32)])],
+      'mobilenum': [null, Validators.compose([Validators.required,Validators.pattern('^[789]+[0-9]{9}$'), Validators.minLength(10), Validators.maxLength(10)])],
+      'regnum': [null, Validators.compose([Validators.required,Validators.pattern('^[0-9]{12}$'), Validators.minLength(12), Validators.maxLength(12)])],
+      'dob': [null, Validators.required],
+      'yop': [null, Validators.compose([Validators.required,Validators.pattern('^[0-9]{4}$')])],
+      'emailid': [null,Validators.compose([Validators.required,Validators.pattern('^([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)')])],
+      'description': [null, Validators.compose([Validators.required, Validators.minLength(30), Validators.maxLength(150)])],
+      'validate': ''
+    });
+  }
+
+  addPost(post) {
+    this.fname = post.fname;
```

Comment:
the method implementation is good. but like as i said, the whole table implementation is wrong. we need to take the value of form and push it into a table data source so that all the entries are shown in table